sunpy/map/mapbase.py

```diff
@@ -1706,11 +1706,15 @@ def rotate(self, angle: u.deg = None, rmatrix=None, order=3, scale=1.0,
         if version.parse(np.__version__) < version.parse("1.20.0") and issubclass(self.data.dtype.type, numbers.Integral) and (missing % 1 != 0):
             warn_user("The specified `missing` value is not an integer, but the data "
                       "array is of integer type, so the output may be strange.")
-        # Pad the image array
-        new_data = np.pad(self.data,
-                          ((pad_y, pad_y), (pad_x, pad_x)),
-                          mode='constant',
-                          constant_values=(missing, missing))
+        try:
+            new_data = np.pad(self.data, 
+                      ((pad_y, pad_y), (pad_x, pad_x)), 
+                      mode='constant', 
+                      constant_values=(missing, missing))
+        except ValueError :
+            error_message = "Unable to pad the array most likely due to the default `missing` keyword as it is set to NaN. Change the `missing` keyword to a different value."
+            warn_user(error_message)
```

Comment:
I was wrong, this should not be a warning, this should still return the error and stop the code from running.

Reply:
So should we do like?
error_message=...........
raise ValueError(error_message)
and update unit test upon it?

Reply:
Yes please.